### DISCUSSION TITLE
Error while starting service on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by blasco403** [GerryDazoo/Slinger#164] [Q&A]
Hello,

I wanted to test running the server locally where I am located instead of on a Windows 10 machine at the source location. I have a Raspberry Pi locally that runs pihole, wireguard VPN back to source location as well as my local Unifi network controller.

I followed the instructions in the README.txt

Python 3.7.3 is installed.

Flask install:

sudo pip install flask
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: flask in /usr/lib/python2.7/dist-packages (1.0.2)

I added the following to: /etc/sysctl.conf 
    net.core.rmem_max = 8192000
    net.core.wmem_max = 8192000

I have rebooted my raspberry pi

I have performed all steps, it appears it is installed correctly. My config.ini and 2 remoteXXX.txt files are located in the slingbox users home directory: The files are a direct copy from the ones that work on my Windows based server.

drwxr-xr-x 2 slingbox slingbox  4096 Oct 22 13:48 .
drwxr-xr-x 4 root     root      4096 Oct 22 11:42 ..
-rw-r--r-- 1 slingbox slingbox   220 Oct 22 11:42 .bash_logout
-rw-r--r-- 1 slingbox slingbox  3523 Oct 22 11:42 .bashrc
-rw-r--r-- 1 root     root       767 Oct 18 13:37 config.ini
-rw-r--r-- 1 slingbox slingbox   807 Oct 22 11:42 .profile
-rw-r--r-- 1 root     root      4254 Oct 19 13:58 remote350.txt
-rw-r--r-- 1 root     root      4254 Oct 19 13:58 remote500.txt
-rw-r--r-- 1 root     root     54680 Oct 19 05:18 slingbox_server.py
-rw-r--r-- 1 root     root       309 Oct 19 05:18 sling.service

I get the following error in sling.log when service starts:

  File "slingbox_server.py", line 365
    nonlocal sid, seq, s_ctl, dbuf, skey, stat, smode
               ^
SyntaxError: invalid syntax
  File "slingbox_server.py", line 365
    nonlocal sid, seq, s_ctl, dbuf, skey, stat, smode
               ^
SyntaxError: invalid syntax
  File "slingbox_server.py", line 365
    nonlocal sid, seq, s_ctl, dbuf, skey, stat, smode
               ^
SyntaxError: invalid syntax
  File "slingbox_server.py", line 365
    nonlocal sid, seq, s_ctl, dbuf, skey, stat, smode
               ^
SyntaxError: invalid syntax
  File "slingbox_server.py", line 365
    nonlocal sid, seq, s_ctl, dbuf, skey, stat, smode
               ^
SyntaxError: invalid syntax

I get this when manually trying to start the service from an SSH console session.

sudo systemctl status sling.service

● sling.service - SlingBox Server Service
   Loaded: loaded (/etc/systemd/system/sling.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sat 2022-10-22 14:00:48 CDT; 9min ago
  Process: 2962 ExecStart=/usr/bin/python -u slingbox_server.py (code=exited, status=1/FAILURE)
 Main PID: 2962 (code=exited, status=1/FAILURE)

Oct 22 14:00:48 mexicopi systemd[1]: sling.service: Service RestartSec=100ms expired, scheduling restart.
Oct 22 14:00:48 mexicopi systemd[1]: sling.service: Scheduled restart job, restart counter is at 5.
Oct 22 14:00:48 mexicopi systemd[1]: Stopped SlingBox Server Service.
Oct 22 14:00:48 mexicopi systemd[1]: sling.service: Start request repeated too quickly.
Oct 22 14:00:48 mexicopi systemd[1]: sling.service: Failed with result 'exit-code'.
Oct 22 14:00:48 mexicopi systemd[1]: Failed to start SlingBox Server Service.

Config.ini

[SLINGBOXES]
sb1=350
sb2=500

[350]
sbtype="350/500/M1"
password=xxxx
ipaddress=192.168.0.4
port=5211
Resolution=12
FrameRate=60
VideoBandwidth=4000
VideoSmoothness=63
VideoSource=1
maxstreams=2
;StartChannel=155
;RemoteLock=yes
Remote=remote350.txt

[500]
sbtype="350/500/M1"
password=xxxx
ipaddress=192.168.0.221
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=4000
VideoSmoothness=63
VideoSource=2
maxstreams=2
;StartChannel=155
;RemoteLock=yes
Remote=remote500.txt
;Remote=testremote.txt

[SERVER]
; local port number for the server to listen on for connections
; port number+1 is used by the remote control code.
port=58048
maxremotestreams=2
URLbase=shaw
enableremote=yes

**Comment by GerryDazoo** [comment]
stop the service and run the command manually. I don't think your running the version of python you think you are. You get this error if you are running python2
/path_to_python -version

**Reply by blasco403** [reply]
Thanks Gerry. It appears I have both Python 2 and 3 installed.

I edited the slingbox.service file and added a 3 to the Python launch command and it is working now.
